src/test/paths.ts

```diff
@@ -165,6 +174,15 @@ describe('paths', () => {
                     }),
                     '/home/felix/mysource/source.php'
                 )
+                // longest prefix matching for server paths
+                assert.strictEqual(
+                    convertDebuggerPathToClient('file:///app/subdir/source.php', {
+                        '/var/www': '/home/felix/mysite',
+                        '/app': '/home/felix/mysource',
+                        '/app/subdir': '/home/felix/mysource/subdir1',
```

Comment:
The list should be built so that without the patch the test fails. Please move `'/app/subdir': '/home/felix/mysource/subdir1',` one line up, before `'/app': '/home...`.